### ISSUE TITLE
supplier address isn't shown in JSON output

### ISSUE THREAD

**Issue by SebastianKonplan** [CycloneDX/cyclonedx-python-lib#712]
A supplier address `PostalAddress` can be added to a `Component` but it isn't printed in the JSON output.

**Comment by jkowalleck** [comment]
could you provide a reproducible example?

**Comment by SebastianKonplan** [comment]
Here is a stripped-down example. The `PostalAddress` isn't output in the `test.json`.
```
#!/usr/bin/env python3
"""The script generates a software bill of materials (SBOM)
in CycloneDX format. A JSON-based CycloneDX is generated."""

import os
import re
from argparse import ArgumentParser
from typing import TYPE_CHECKING

try:
    from cyclonedx.factory.license import LicenseFactory
    from cyclonedx.model import XsUri
    from cyclonedx.model.bom import Bom
    from cyclonedx.model.component import Component, ComponentType
    from cyclonedx.model.contact import OrganizationalEntity
    from cyclonedx.model.contact import OrganizationalContact
    from cyclonedx.model.contact import PostalAddress
    from cyclonedx.output.json import JsonV1Dot5
    from cyclonedx.schema import SchemaVersion
    from cyclonedx.validation.json import JsonStrictValidator
except ImportError as exc:
    raise ImportError("Install missing package with: " +
                      "\"pip install 'cyclonedx-python-lib[validation]" +
                      "'\"") from exc

if TYPE_CHECKING:
    from cyclonedx.output.json import Json as JsonOutputter

def main():
    bom = Bom()

    bom.metadata.component = root_component = Component(
        name='myname',
        type=ComponentType.OPERATING_SYSTEM,
        bom_ref='myref',
        supplier=OrganizationalEntity(
            name='my company',
            urls=[XsUri('http://www.mycompany.com')],
            contacts=[OrganizationalContact(
                phone='+49 (30) 123 456 78',
                email='info@mycompany.com',
            )],
            address=PostalAddress(  # FIXME: isn't shown in json?
                country='Germany',
                postal_code='D-12345',
                locality='City',
                street_address='Street 123',
            ),
        ),
    )

    # Create CycloneDX
    my_json_outputter: 'JsonOutputter' = JsonV1Dot5(bom)
    my_json_outputter.output_to_file(
        filename='test.json',
        allow_overwrite=True,
        indent=2)

    serialized_json = my_json_outputter.output_as_string(indent=2)
    # print(serialized_json)

    # Validation
    my_json_validator = JsonStrictValidator(SchemaVersion.V1_6)

    validation_errors = my_json_validator.validate_str(serialized_json)

    if validation_errors:
        raise ValueError("JSON invalid, ValidationError:"
                         + repr(validation_errors))


if __name__ == "__main__":
    main()

```

**Comment by jkowalleck** [comment]
re: https://github.com/CycloneDX/cyclonedx-python-lib/issues/712#issuecomment-2419656938

result in `test.json`:
```json
{
  "dependencies": [
    {
      "ref": "myref"
    }
  ],
  "metadata": {
    "component": {
      "bom-ref": "myref",
      "name": "myname",
      "supplier": {
        "contact": [
          {
            "email": "info@mycompany.com",
            "phone": "+49 (30) 123 456 78"
          }
        ],
        "name": "my company",
        "url": [
          "http://www.mycompany.com"
        ]
      },
      "type": "operating-system"
    },
    "timestamp": "2024-10-21T08:39:29.767068+00:00"
  },
  "serialNumber": "urn:uuid:b0d4e013-3b4d-4083-a22a-985499e5094f",
  "version": 1,
  "$schema": "http://cyclonedx.org/schema/bom-1.5.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.5"
}
```

**Comment by jkowalleck** [comment]
`OrganizationalContact.address` is not available before CycloneDX 1.6.

your example code is rendering to CycloneDX 1.5
It is expected to not see any address in there.

after altering your example to use `from cyclonedx.output.json import JsonV1Dot6`, here is the actual result:
```json
{
  "dependencies": [
    {
      "ref": "myref"
    }
  ],
  "metadata": {
    "component": {
      "bom-ref": "myref",
      "name": "myname",
      "supplier": {
        "address": {
          "country": "Germany",
          "locality": "City",
          "postalCode": "D-12345",
          "streetAddress": "Street 123"
        },
        "contact": [
          {
            "email": "info@mycompany.com",
            "phone": "+49 (30) 123 456 78"
          }
        ],
        "name": "my company",
        "url": [
          "http://www.mycompany.com"
        ]
      },
      "type": "operating-system"
    },
    "timestamp": "2024-10-21T08:51:32.063009+00:00"
  },
  "serialNumber": "urn:uuid:91362a54-91e8-4f0e-811b-60afba7c2d1c",
  "version": 1,
  "$schema": "http://cyclonedx.org/schema/bom-1.6.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.6"
}
```

result contains the address.

conclusion, all as expected.

**Comment by SebastianKonplan** [comment]
@jkowalleck Thank you very much for the explanation.
